Dockerfile

```diff
@@ -0,0 +1,25 @@
+# Stage 1: Build stage
+FROM python:3.8-slim AS builder
```

Comment:
Consider adding an `ARG` for the Python base image version to allow flexibility and better control over the environment. This can be done by adding a line like `ARG PYTHON_VERSION=3.8-slim` and then using it in the `FROM` directive as `FROM python:${PYTHON_VERSION} AS builder`.